internal/operators/operators.go

```diff
@@ -0,0 +1,258 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operators
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	columnhelpers "github.com/inspektor-gadget/inspektor-gadget/internal/column-helpers"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+type Runner interface {
+	ID() string
+	Columns() columnhelpers.Columns
+	Gadget() gadgets.Gadget
+	Context() context.Context
+	Operators() Operators
+	Logger() logger.Logger
+}
+
+type Operator interface {
+	// Name must return a unique name for the operator
+	Name() string
+
+	// Description is an optional description to show to the user
+	Description() string
+
+	// Params will return global params (required) for this operator
+	Params() params.Params
+
+	// PerGadgetParams will return params (required) per gadget instance of the operator
+	PerGadgetParams() params.Params
+
+	// Dependencies can list other operators that this operator depends on
+	Dependencies() []string
+
+	// CanOperateOn should test whether the operator supports the given gadget. Init has not
+	// necessarily been called at this point.
+	CanOperateOn(gadgets.Gadget) bool
+
+	// Init allows the operator to initialize itself
+	Init(params params.Params) error
+
+	// Cleanup allows the operator to clean up stuff prior to exiting
+	Cleanup() error
+
+	// PreGadgetRun is called before a gadget is started; the operator must return something that implements operator
+	// This is useful to create a context for an operator by wrapping it.
+	// Params given here are the ones returned by PerGadgetParams()
+	PreGadgetRun(Runner, any, params.Params) (Operator, error)
+
+	// PostGadgetRun is called on the operator that was returned from PrepareTrace after a
+	// gadget was stopped
+	PostGadgetRun() error
+
+	// EnrichEvent is called on the operator returned by StartTrace and should perform
+	// the actual enrichment
+	EnrichEvent(any) error
+}
+
+type Operators []Operator
+
+// KubernetesFromMountNSID is a typical kubernetes operator interface that adds node, pod, namespace and container
+// information given the MountNSID
+type KubernetesFromMountNSID interface {
+	ContainerInfoSetters
+	GetMountNSID() uint64
+}
+
+type ContainerInfoSetters interface {
+	SetContainerInfo(pod, namespace, container string)
+	SetNode(string)
+}
+
+var operators = map[string]Operator{}
+
+type operatorWrapper struct {
+	Operator
+	initOnce    sync.Once
+	initialized bool
+}
+
+func (e *operatorWrapper) Init(params params.Params) (err error) {
+	e.initOnce.Do(func() {
+		err = e.Operator.Init(params)
+		e.initialized = true
+	})
+	return err
+}
+
+// RegisterOperator adds a new operator to the registry
+func RegisterOperator(operator Operator) {
```

Comment:
I think the function names in this package are a little bit redundant. We are calling it `operators.RegisterOperators(myOperator)`. It could simply be `operators.Register(myOperator)`.

Check [these best practices](https://google.github.io/styleguide/go/best-practices#function-and-method-names).